agent/api/task/task.go

```diff
@@ -2419,7 +2458,7 @@ func (task *Task) dockerPortMap(container *apicontainer.Container, dynamicHostPo
 				return nil, err
 			}
 
-			// append ranges to the dockerPortMap
+			// For the ContainerPortRange case, append ranges to the dockerPortMap.
 			// nat.ParsePortSpec returns a list of port mappings in a format that docker likes
 			mappings, err := nat.ParsePortSpec(hostPortRange + ":" + containerPortRange + "/" + protocol)
 			if err != nil {
```

Comment:
what do we do with this error (or do we just throw it away)?

Reply:
Example errors that might be returned from [ParsePortSpec](https://github.com/docker/go-connections/blob/58542c764a1173ea3dac965d89146c931a2946f7/nat/nat.go#L171) are

1. Invalid ip address <ip>: <error message>
2. No port specified: <raw port><empty>
3. Invalid containerPort: <container port>
4. Invalid hostPort: <host port>
5. Invalid ranges specified for container and host Ports: <container port> and <host port>
6. Invalid proto: protocol

Reply:
When an error is returned from [task.dockerPortMap](https://github.com/aws/amazon-ecs-agent/blob/4943e4d853a84cf4d76b9d1241ee1f4f189b9c4a/agent/api/task/task.go#L1867-L1870), the container will fail to create. And depends on whether the container is an essential container or not, the task be stopped if it is an essential container.

Reply:
offline conversation: the error will be surfaced via logs prior to this.